### DISCUSSION TITLE
Datatable ResponsiveLayout Stack not stacking output on smaller screens

### DISCUSSION THREAD

**Discussion by chrishj59** [primefaces/community#1056] [PrimeReact] [answered]
I have a fairly wide table that has 9 columns. It displays horizontally fine on a larger screen MacBook.
When using the option responsiveLayout="stack" it does not reflow when simulating smaller width devices using FireFox Responsive Design Mode with size under 964x926. Under this size the titles are not visible.
If I use the option responsiveLayout="scroll" it is possible to still scroll through the column headings.

In am using version primeReact 10.3.3.

This can be seen in the documentation [codeSandBox](https://codesandbox.io/s/8pln2h?file=/src/App.jsx)

The documentation says responsiveLayout has been deprecated after 9.2.0. 
Is there another option I should be using?

Thanks

**Comment by melloware** [comment, marked as answer]
Stack has been removed.  There is other option in PrimeReact.  However this user did recreate the old Stack Mode in PR 10 if you want to use his code: https://github.com/primefaces/primereact/issues/4078#issuecomment-1751819234

https://codesandbox.io/s/responsive-datatable-yyjsp4

The main code is in files datatable.js and datatable.css.

**Reply by chrishj59** [reply]
What is the recommended approach ?
Many thanks.

**Reply by melloware** [reply]
Using that user's code...

**Comment by baracuda69** [comment]
For anyone trying to make responsive tables using Prime DataTable in unstyled mode:

1. Add responsive-layout="stack" attribute to DataTabe
This will ADD <span data-pc-section="columntitle">...</span> to each table cell <td>

2. Use the following scss

/* responsive-table.scss */
[data-pc-section="columntitle"] {
    display: none;
}

@media screen and (max-width: 960px) {
    table:has([data-pc-section="columntitle"]) {
        min-width: 300px !important;
        display: block;

        thead {
            tr {
                display: none;
            }
        }

        tbody {
            display: block;

            tr {
                display: block;
                border-radius: 6px;
                margin-bottom: 1em;
                position: relative;
                z-index: auto;
                margin-top: 1.4rem;

                td {
                    border: none;
                    display: flex;
                    flex-direction: row;
                    justify-content: space-between;
                    align-items: center;
                    overflow-wrap: break-word;

                    [data-pc-section="columntitle"] {
                        display: inline;
                    }
                }
            }
        }
    }
}
